.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+---
+# Run the checkrefs.py script and check for typos.
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+jobs:
+  python-script:
+    name: 'Checkrefs.py: Actors, Templates and Maps'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get latest release tag of 0 A.D.
+        id: tag_name
+        run: |
+          echo "release_tag=$(curl -s https://api.github.com/repos/0ad/0ad/releases/latest | jq -r '.tag_name')" >> $GITHUB_OUTPUT
+      - name: Check out 0 A.D. (${{ steps.tag_name.outputs.release_tag }})
+        uses: actions/checkout@v3
+        with:
+          repository: 0ad/0ad
```

Comment:
Instead, let's download the 0ad tar.gz (https://github.com/0ad/0ad/tags) and implement [caching](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#managing-caches)

There's a small example at https://github.com/curl/curl/pull/10393/files that fits our situation perfectly.

Reply:
With caching, I got very similar speeds of around 3~5mins.
<img width="901" alt="SCR-20230302-rbpd" src="https://user-images.githubusercontent.com/92653266/222520815-e630844a-242b-47f4-bc3b-3dc999a9c5e0.png">

The speed of getting the cache was almost the same as a new checkout:
<img width="508" alt="SCR-20230302-rcos" src="https://user-images.githubusercontent.com/92653266/222521500-fb131c0c-3045-4c31-a4fa-9f52c1770940.png">

```yml
    steps:
      - name: Get latest release tag of 0 A.D.
        id: tag_name
        run: |
          echo "release_tag=$(curl -s https://api.github.com/repos/0ad/0ad/releases/latest | jq -r '.tag_name')" >> $GITHUB_OUTPUT
      - name: Check if there is a cached version of 0 A.D. (${{ steps.tag_name.outputs.release_tag}})
        id: cache-0ad_directory
        uses: actions/cache@v3
        with:
          path: 0ad_directory
          key: ${{ runner.os }}-0ad_directory-${{ steps.tag_name.outputs.release_tag}}
      - name: Check out 0 A.D. if there is no cached version
        if: steps.cache-0ad_directory.outputs.cache-hit != 'true'
        uses: actions/checkout@v3
        with:
          repository: 0ad/0ad
          path: 0ad_directory
          ref: ${{ steps.tag_name.outputs.release_tag }}
          clean: true
```

Try to beat 3-5mins!